src/components/LetterDetail.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useRef, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { CloudIcon, ColorSirenIcon, SnowIcon, ThermostatIcon, WarmIcon } from '@/assets/icons';
+
+import ReportModal from './ReportModal';
+
+export default function LetterDetail({ title, text }: { title: string; text: string }) {
+  const FONT = '';
+  const THEME = 'celebrate';
+  const DEGREES = [
+    { icon: WarmIcon, title: '따뜻해요' },
+    { icon: CloudIcon, title: '그럭저럭' },
+    { icon: SnowIcon, title: '앗! 차가워' },
+  ];
+  const [degreeModalOpen, setDegreeModalOpen] = useState<boolean>(false);
+  const [reportModalOpen, setReportModalOpen] = useState<boolean>(false);
+
+  const degreeButtonRef = useRef<HTMLButtonElement>(null);
+  const handleOutsideClick = (event: MouseEvent) => {
+    const target = event.target as Node;
+    if (!target || degreeButtonRef.current?.contains(target)) {
+      return;
+    }
+    setDegreeModalOpen(false);
+  };
+  useEffect(() => {
+    document.body.addEventListener('click', handleOutsideClick);
+
+    return () => {
+      document.body.removeEventListener('click', handleOutsideClick);
+    };
+  }, []);
+  return (
+    <>
+      {reportModalOpen && <ReportModal onClose={() => setReportModalOpen(false)} />}
+      <div className={twMerge(`flex grow flex-col gap-3 px-5 pb-7.5`, THEME)}>
+        <div className="absolute top-5 right-5 flex gap-3">
+          <button
+            ref={degreeButtonRef}
+            className="flex items-center justify-center gap-1"
+            onClick={() => {
+              setDegreeModalOpen((cur) => !cur);
+            }}
+          >
+            <img src={ThermostatIcon} alt="편지 온도 아이콘" />
+            <span className="caption-b text-primary-1">편지 온도</span>
+          </button>
+          <button
+            onClick={() => {
+              setReportModalOpen(true);
+            }}
+          >
+            <img src={ColorSirenIcon} alt="신고 아이콘" />
+          </button>
+          {degreeModalOpen && (
+            <div className="caption-b text-primary-1 bg-primary-5 absolute top-7 z-40 flex flex-col gap-1 p-2 shadow">
+              {DEGREES.map((degree, idx) => {
+                return (
+                  <button
+                    key={idx}
+                    className="flex items-center justify-start gap-1"
+                    onClick={() => {
+                      console.log(idx);
+                    }}
+                  >
+                    <img src={degree.icon} alt={`${degree.title} 아이콘`} /> {degree.title}
+                  </button>
+                );
+              })}
```

Comment:
이것도 취향차이겠지만! 이렇게 하면 return 키워드 사용없이도 가능합니당

```suggestion
              {DEGREES.map((degree, idx) => (
                  <button
                    key={idx}
                    className="flex items-center justify-start gap-1"
                    onClick={() => {
                      console.log(idx);
                    }}
                  >
                    <img src={degree.icon} alt={`${degree.title} 아이콘`} /> {degree.title}
                  </button>
                )
              )}
```

Reply:
오홍 return 넣는거 귀찮아서 이 방법도 한번 고려해봐야겠습니다!!!